src/accumulator/accumulated_map.rs

```diff
@@ -7,6 +7,11 @@ use super::{accumulated::Accumulated, Accumulator, AnyAccumulated};
 #[derive(Default, Debug)]
 pub struct AccumulatedMap {
     map: FxHashMap<IngredientIndex, Box<dyn AnyAccumulated>>,
+    /// `true` if any query called by this query has any accumulated values.
+    ///
+    /// Tracking whether any dependency has any accumulated values helps
+    /// skipping over entire subtrees when collecting accumulated values.
+    dependency_with_accumulated_values: bool,
```

Comment:
Small nit: consider making this an enum instead? It'd make the callsites a bit more legible.

Reply:
This is a great suggestion. Thanks